### PR TITLE
frontend: Home: Show the origin of each cluster

### PR DESCRIPTION
**Pull request by skoeva**
This change introduces a new "Origin" column for the cluster list on the home page, displaying whether a cluster came from a kubeconfig, Headlamp, or a dynamic cluster, with references when available.

Fixes: #1632 

### Testing
- [X] Open Headlamp with 2+ active clusters
- [X] Ensure that the cluster list on the Home page includes the Origin column with respective origins

image

### CODE REVIEW

File: frontend/src/components/App/Home/index.tsx [hunk 1]
```diff
@@ -209,6 +209,17 @@ function useWarningSettingsPerCluster(clusterNames: string[]) {
   return warningLabels;
 }
 
+function getOrigin(cluster: Cluster) {
```
Comment (illume) [inline frontend/src/components/App/Home/index.tsx:212]: Please add function documentation?

File: frontend/src/components/App/Home/index.tsx [hunk 2]
```diff
@@ -209,6 +209,17 @@ function useWarningSettingsPerCluster(clusterNames: string[]) {
...
+  if (cluster.meta_data?.source === 'kubeconfig') {
+    return `Kubeconfig (${process.env.KUBECONFIG ?? '~/.kube/config'})`;
+  } else if (cluster.meta_data?.source === 'dynamic_cluster') {
+    return `Dynamic cluster`;
+  } else if (cluster.meta_data?.source === 'in_cluster') {
+    return 'Headlamp';
```
Comment (illume) [inline frontend/src/components/App/Home/index.tsx:218]: It could be Headlamp is branded as something else, so we shouldn't use Headlamp here I think.

I think "In-Cluster" is maybe better?

**Review by illume — COMMENTED** [review]
👍 thanks

Left a couple of notes...

File: frontend/src/components/App/Home/index.tsx (hunk 2 above)
Comment (skoeva) [reply frontend/src/components/App/Home/index.tsx]: done

File: frontend/src/components/App/Home/index.tsx [hunk 3]
```diff
@@ -264,6 +281,13 @@ function HomeComponent(props: HomeComponentProps) {
                   </Link>
                 ),
               },
+              {
+                label: t('Origin'),
+                getValue: cluster => getOrigin(cluster),
+                render: ({ name }) => (
+                  <Typography variant="body2">{getOrigin(clusters[name])}</Typography>
```
Comment (illume) [inline frontend/src/components/App/Home/index.tsx:288]: Oh... I guess these will need translations.

File: frontend/src/components/App/Home/index.tsx (hunk 1 above)
Comment (illume) [reply frontend/src/components/App/Home/index.tsx:218]: Maybe the return type could be added `function getOrigin(cluster: Cluster): string {` instead of in the returns.

`@returns the description of where the cluster is picked up from. Dynamic, in-cluster or from a kubeconfig file`

File: frontend/src/components/App/Home/index.tsx [hunk 4]
```diff
@@ -209,6 +209,23 @@ function useWarningSettingsPerCluster(clusterNames: string[]) {
...
 
+/**
+ * Gets the origin of a cluster.
+ *
+ * @param {Cluster} cluster
+ * @returns {string}
+ */
+function getOrigin(cluster: Cluster) {
+  if (cluster.meta_data?.source === 'kubeconfig') {
+    return `Kubeconfig (${process.env.KUBECONFIG ?? '~/.kube/config'})`;
```
Comment (illume) [inline frontend/src/components/App/Home/index.tsx:220]: But maybe the translations could be added in this function, so they can be detected by that i18n script.

**Review by illume — APPROVED** [review]
👍 thanks

(probably just the translation change should be done, the docs change is just a nit)

File: frontend/src/components/App/Home/index.tsx (hunk 1 above)
Comment (skoeva) [reply frontend/src/components/App/Home/index.tsx]: done

File: frontend/src/components/App/Home/index.tsx (hunk 4 above)
Comment (skoeva) [reply frontend/src/components/App/Home/index.tsx]: added translations

File: frontend/src/components/App/Home/index.tsx [hunk 5]
```diff
@@ -236,6 +236,24 @@ function HomeComponent(props: HomeComponentProps) {
...
+  /**
+   * Gets the origin of a cluster.
+   *
+   * @param cluster
+   * @returns A description of where the cluster is picked up from: dynamic, in-cluster, or from a kubeconfig file.
+   */
+  function getOrigin(cluster: Cluster): string {
+    if (cluster.meta_data?.source === 'kubeconfig') {
+      const kubeconfigPath = process.env.KUBECONFIG ?? '~/.kube/config';
+      return t('translation|Kubeconfig') + ` (${kubeconfigPath})`;
```
Comment (joaquimrocha) [inline frontend/src/components/App/Home/index.tsx:248]: We don't need to translate Kubeconfig. And the value here should be: `kubeconfig:PATH_TO_FILE`
Can you make another PR to fix this, since this one is merged?

File: frontend/src/components/App/Home/index.tsx [hunk 6]
```diff
@@ -236,6 +236,24 @@ function HomeComponent(props: HomeComponentProps) {
...
+    } else if (cluster.meta_data?.source === 'dynamic_cluster') {
+      return t('translation|Dynamic cluster');
```
Comment (joaquimrocha) [inline frontend/src/components/App/Home/index.tsx:250]: I am not sure this will give a lot of info to the user. I'd prefer the still poor but more accurate `Plugin` for now.

**Review by joaquimrocha — COMMENTED** [review]
I think a few things can be improved.

File: frontend/src/components/App/Home/index.tsx (hunk 6 above)
Comment (skoeva) [reply frontend/src/components/App/Home/index.tsx:250]: Got it

File: frontend/src/components/App/Home/index.tsx (hunk 5 above)
Comment (skoeva) [reply frontend/src/components/App/Home/index.tsx:248]: This change as well as the other one mentioned are done here: #2543
